spec/models/submit_spec.rb

```diff
@@ -114,5 +114,32 @@ def add_participant(contest_owner = owner)
       expect(contest_solutions[0][:solutions][0][:attempts]).to eq(1)
       expect(contest_solutions[1][:solutions][0][:attempts]).to eq(1)
     end
+
+    it 'sorts problems in order: required, created_at' do
```

Comment:
zmieniłbym na `sorts problems in order: required DESC, created_at DESC` lub `sorts problems in order: required before optional , created_at DESC`